### ISSUE TITLE
spin plugin upgrade should understand version 0.10.0 is greater than version 0.9.1

### ISSUE THREAD

**Issue by chikoski** [fermyon/spin#2954]
I tried to upgrade the cloud plugin to 0.10.0, but the plugin update command stopped with the following error message:

```zsh
$ spin plugin upgrade cloud
Error: Newer version 0.9.1 of plugin 'cloud' is already installed. To downgrade to version 0.10.0, run `spin plugins upgrade` with the `--downgrade` flag.
```

The command should recognize the version number correctly and enable us to upgrade the plugin to 0.10.0 from 0.9.1 without setting `--downgrade` flag. 

* Spin version: spin 3.0.0 (737778e 2024-11-07)
* Installed plugin versions (`spin plugins list --installed`)
  - cloud 0.9.1 [installed]
  - js2wasm 0.6.1 [installed]

**Comment by itowlson** [comment]
Thanks for reporting this.  It was fixed via #2947 but that's not in a release yet.  Sorry for the inconvenience.

**Comment by chikoski** [comment]
Thanks for letting me know that the issue has already been fixed. Let me close the issue.
